### PR TITLE
backupccl: update env vars in TestPutAWSKMSEndpoint and TestPutS3Endpoint

### PR DESCRIPTION
**Pull request by msbutler**
TestPutAWSKMSEndpoint and TestPutS3Endpoint have been ignored in our cloud nightly unit test job because they were incorrectly calling for env vars that didn't exist on the cluster. This patch updates the env vars required for these tests.

Release note: None

Epic: None

### CODE REVIEW

**Comment by msbutler** [comment]
on my [latest custom build](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7285948?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true) run, TestPutAWSKMSEndpoint/disallow-endpoints did not surface the expected error because the `AWSEndpointParam` does not get set. Given how this test was initially designed, it seems like it expects an env var on the machine running to the test to store the endpoint. Should we ask dev inf to add a new env var, and ignore this sub test for now while we we wait for them? Or, should we take the tack of `TestS3DisallowImplicitCredentials`, and add a phony endpoint to the unit test?

**Review by rhu713 — DISMISSED** [review]

File: pkg/cloud/amazon/aws_kms_test.go [hunk 1]
```diff
@@ -216,15 +216,18 @@ func TestPutAWSKMSEndpoint(t *testing.T) {
...
+	expectedParams := []string{
+		AWSEndpointParam,
+		AWSAccessKeyParam,
+		KMSRegionParam,
+		AWSSecretParam}
+	for _, param := range expectedParams {
+		env := nightlyEnvVarS3Params[param]
 		v := os.Getenv(env)
 		if v == "" {
+			if param == AWSEndpointParam {
```
Comment (rhu713) [inline pkg/cloud/amazon/aws_kms_test.go:228]: Discussed in slack: we can add the standard S3 endpoint as the environment variable in teamcity for now.

**Comment by msbutler** [comment]
The latest failure on `TestPutAWSKMSEndpoint/allow-endpoints/simple_encrypt_decrypt` may actually be legit, though I don't know enough about kms shenanigans to be sure. Should a kms uri contain an aws enpoint with the `AWS_ENDPOINT` key?
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7400142?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

```
Failed
=== RUN   TestPutAWSKMSEndpoint/allow-endpoints/simple_encrypt_decrypt
    kms_test_utils.go:73:
          Error Trace:  /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3861/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/cloud/amazon/amazon_test_/amazon_test.runfiles/com_github_cockroachdb_cockroach/pkg/cloud/amazon/kms_test_utils.go:73
          Error:        Received unexpected error:
                        SerializationError: failed to unmarshal response error
                          status code: 400, request id: 6HE276ZVRGGFS4ZW
                        caused by: UnmarshalError: failed decoding error message
                          00000000  3c 3f 78 6d 6c 20 76 65  72 73 69 6f 6e 3d 22 31  |<?xml version="1|
                        00000010  2e 30 22 20 65 6e 63 6f  64 69 6e 67 3d 22 55 54  |.0" encoding="UT|
                        00000020  46 2d 38 22 3f 3e 0a 3c  45 72 72 6f 72 3e 3c 43  |F-8"?>.<Error><C|
                        00000030  6f 64 65 3e 49 6e 76 61  6c 69 64 52 65 71 75 65  |ode>InvalidReque|
                        00000040  73 74 3c 2f 43 6f 64 65  3e 3c 4d 65 73 73 61 67  |st</Code><Messag|
                        00000050  65 3e 4d 69 73 73 69 6e  67 20 72 65 71 75 69 72  |e>Missing requir|
                        00000060  65 64 20 68 65 61 64 65  72 20 66 6f 72 20 74 68  |ed header for th|
                        00000070  69 73 20 72 65 71 75 65  73 74 3a 20 78 2d 61 6d  |is request: x-am|
                        00000080  7a 2d 63 6f 6e 74 65 6e  74 2d 73 68 61 32 35 36  |z-content-sha256|
                        00000090  3c 2f 4d 65 73 73 61 67  65 3e 3c 52 65 71 75 65  |</Message><Reque|
                        000000a0  73 74 49 64 3e 36 48 45  32 37 36 5a 56 52 47 47  |stId>6HE276ZVRGG|
                        000000b0  46 53 34 5a 57 3c 2f 52  65 71 75 65 73 74 49 64  |FS4ZW</RequestId|
                        000000c0  3e 3c 48 6f 73 74 49 64  3e 42 43 34 4e 65 42 4a  |><HostId>BC4NeBJ|
                        000000d0  46 63 66 69 5a 46 4f 7a  66 45 71 47 4d 43 63 31  |FcfiZFOzfEqGMCc1|
                        000000e0  6e 4c 4c 69 43 74 52 63  35 56 67 42 74 5a 77 31  |nLLiCtRc5VgBtZw1|
                        000000f0  4a 33 68 45 49 71 31 73  50 68 2b 77 6d 30 44 62  |J3hEIq1sPh+wm0Db|
                        00000100  4c 44 31 43 35 61 59 49  72 68 53 49 73 36 77 30  |LD1C5aYIrhSIs6w0|
                        00000110  62 57 45 30 3d 3c 2f 48  6f 73 74 49 64 3e 3c 2f  |bWE0=</HostId></|
                        00000120  45 72 72 6f 72 3e                                 |Error>|
                        caused by: invalid character '<' looking for beginning of value
          Test:         TestPutAWSKMSEndpoint/allow-endpoints/simple_encrypt_decrypt
        --- FAIL: TestPutAWSKMSEndpoint/allow-endpoints/simple_encrypt_decrypt (0.67s)
```

**Comment by msbutler** [comment]
@adityamaru could you take a look at this test failure?

**Comment by msbutler** [comment]
RFAL! latest  nightly is all green

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7552339?showRootCauses=false

**Comment by msbutler** [comment]
unrelated flake, rerunning

**Review by adityamaru — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r=adityamaru

**Pull request merged**